### PR TITLE
Favourite tab API integration

### PR DESCRIPTION
**Pull request by arjun-m06**
* Api integration for favorite products in supplements and recourses popup
* Props to support in recommendation, products swap and favorite pages.
* Scroll pagination added.

### CODE REVIEW

File: components/NewSupplementsModal/FavoritesSupplements.tsx [hunk 1]
```diff
@@ -1,11 +1,172 @@
...
+import { AS_NEEDED, DEFAULT_PRODUCT_QUANTITY, EMPTY, EVERY_MORNING, ITEMS_PER_PAGE, SUPPLIER } from "@/lib/const";
+import { ProductAndResource } from "./CatalogInterface";
+interface favoriteProps {
+  favoriteProductCallBack?: any;
+  selectedSupplementIds?: string[];
+  handleSupplementSelection?: (productDetails: ProductAndResource | any) => void;
+}
+
+function FavoritesSupplements({ favoriteProductCallBack, selectedSupplementIds, handleSupplementSelection }: favoriteProps) {
+  const [getFilterField, setFilterField] = useState<ORDER>(ORDER.CREATED_AT);
```
[the remark this answers is not in the document]
Comment (arjun-m06) [reply components/NewSupplementsModal/FavoritesSupplements.tsx:17]: done

File: components/NewSupplementsModal/FavoritesSupplements.tsx [hunk 2]
```diff
@@ -1,11 +1,172 @@
...
+  };
+
+  useEffect(() => {
+    let productData: any[] = [];
+    var getBrandName = (brand: any) => {
+      return brand?.map((brandName: any) => brandName.name).toString().toUpperCase()
+    }
+    defaultFavoritesProductsAPI?.favouriteProducts?.edges?.map((productItem: any) => {
+      let varientId = EMPTY;
+      let vName = EMPTY;
```
[the remark this answers is not in the document]
Comment (arjun-m06) [reply components/NewSupplementsModal/FavoritesSupplements.tsx:58]: variantName - given

File: components/NewSupplementsModal/FavoritesSupplements.tsx [hunk 3]
```diff
@@ -1,11 +1,172 @@
...
+      let price = 0;
+      let currency = EMPTY;
+      let dosage = EMPTY;
+      let qtyAvailable = 0;
+      let variantStatus = [EMPTY]
+      productItem?.node?.variants
+        ?.filter((variantItem: any, index: number) => index === 0)
```
[the remark this answers is not in the document]
Comment (arjun-m06) [reply components/NewSupplementsModal/FavoritesSupplements.tsx:65]: i removed .. unused

File: components/utils/utils.ts [hunk 4]
```diff
@@ -106,31 +106,48 @@ export function ProductData(productsApiData:any){
...
-      price: price,
-      note: EMPTY,
-      overview: EMPTY,
-      varientName: vName,
-      description: EMPTY,
-      unit: "1",
-      numberOfDay: EMPTY,
-      period: EMPTY,
-      variants: productItem.node.variants,
+      let vName = EMPTY;
```
[the remark this answers is not in the document]
Comment (arjun-m06) [reply components/utils/utils.ts:109]: changed

File: components/NewSupplementsModal/FavoritesSupplements.tsx [hunk 5]
```diff
@@ -1,11 +1,172 @@
...
-function FavoritesSupplements(){
-    return(
-        <>
+  // Changing varient and logics to change the price and id , depending on the product varient dropdown
+  const onSelectItem = (id: string) => {
+    const productData: any[] = updateSelectedVariant(supplementsAPI, id);
+    setSupplementsAPI(productData);
+  };
+
+  const updateSelectedVariant = (currentProducts: any[], id: string) => {
```
[the remark this answers is not in the document]
Comment (arjun-m06) [reply components/NewSupplementsModal/FavoritesSupplements.tsx:119]: Will create a context at that time its fixed
already in issue sheet

File: components/NewSupplementsModal/CatalogProducts.tsx [hunk 6]
```diff
@@ -94,7 +94,7 @@ export function CatalogProducts({
             </div>
             <p className="font-semibold text-xsm">{productName}</p>
             <p className="font-semibold text-xsm min-h-[15px]">{brand}</p>
-            <p className="text-semi-bold">{quantity}</p>
+            <p className="text-semi-bold">{quantity} Items in stock</p>
```
[the remark this answers is not in the document]
Comment (arjun-m06) [reply components/NewSupplementsModal/CatalogProducts.tsx]: done

**Review by visak-kumar — APPROVED** [review]

**Review by joneskebhart9787 — APPROVED** [review]

**Pull request merged**